### PR TITLE
feat: Added pvc accessModes support

### PR DESCRIPTION
**Pull request by dandawg**
# What this PR does / why we need it:

This PR updates the feast services structure to allow for users to use a valid PVC accessModes list of their choosing. The default is set to ["ReadWriteOnce"], for cases where a PVC config is included in the spec, but the accessModes field is not included. 

Previously, the PVC accessModes list was hard-coded to ["ReadWriteMany"] (for each of the Feast services where PVCs were used).

# Which issue(s) this PR fixes:

#4853 

# Misc

### CODE REVIEW

**Review by tchughesiv — COMMENTED** [review]
please create a GH issue for this change

**Review by tchughesiv — COMMENTED** [review]
please link the relevant GH issue

File: infra/feast-operator/internal/controller/featurestore_controller_pvc_test.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ var _ = Describe("FeatureStore Controller-Ephemeral services", func() {
 		onlineStoreMountPath := "/online"
 		registryMountPath := "/registry"
 
+		accessModes := []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}
```
Comment (tchughesiv) [inline infra/feast-operator/internal/controller/featurestore_controller_pvc_test.go:72]: can we test setting multiples here?
```suggestion
		accessModes := []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce, corev1.ReadWriteMany}
```

**Comment by dandawg** [comment]
> please link the relevant GH issue

Done. Thanks for the reminder.

File: infra/feast-operator/api/v1alpha1/featurestore_types.go [hunk 2]
```diff
@@ -249,6 +249,8 @@ type PvcConfig struct {
 // The PVC name is the same as the associated deployment name.
 // +kubebuilder:validation:XValidation:rule="self == oldSelf",message="PvcCreate is immutable"
 type PvcCreate struct {
+	// AccessModes k8s persistent volume access modes. Default []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}.
```
[the remark this answers is not in the document]
Comment (dandawg) [reply infra/feast-operator/api/v1alpha1/featurestore_types.go]: Done

File: infra/feast-operator/internal/controller/featurestore_controller_pvc_test.go (hunk 1 above)
Comment (dandawg) [reply infra/feast-operator/internal/controller/featurestore_controller_pvc_test.go]: Done and test is passing with two.

File: infra/feast-operator/internal/controller/services/util.go [hunk 3]
```diff
@@ -94,6 +94,9 @@ func ApplyDefaultsToStatus(cr *feastdevv1alpha1.FeatureStore) {
 				pvc := services.Registry.Local.Persistence.FilePersistence.PvcConfig
 				if pvc.Create != nil {
```
Comment (tchughesiv) [inline infra/feast-operator/internal/controller/services/util.go:95]: i wonder if this whole `pvc.Create` section would work as a separate function that all 3 service types could call?

Comment (dandawg) [reply infra/feast-operator/internal/controller/services/util.go]: I added a function to consolidate this bit.

File: infra/feast-operator/internal/controller/services/util.go [hunk 4]
```diff
@@ -182,6 +176,24 @@ func ensureRequestedStorage(resources *v1.VolumeResourceRequirements, requestedS
...
+func ensurePVCDefaults(pvc *feastdevv1alpha1.PvcConfig, feastType FeastServiceType) {
+	var storageRequest string
+	switch feastType {
+	case OnlineFeastType:
+		storageRequest = DefaultOnlineStorageRequest
+	case OfflineFeastType:
+		storageRequest = DefaultOfflineStorageRequest
+	case RegistryFeastType:
+		storageRequest = DefaultRegistryStorageRequest
+	}
```
Comment (tchughesiv) [inline infra/feast-operator/internal/controller/services/util.go:188]: maybe `storageRequest` could be an input instead?
```suggestion
func ensurePVCDefaults(pvc *feastdevv1alpha1.PvcConfig, feastType FeastServiceType, storageRequest string) {
```

**Review by tchughesiv — COMMENTED** [review]
one nit, otherwise lgtm

**Review by tchughesiv — APPROVED** [review]

**Pull request merged**